File: Api/Domain/Models/Room.cs
namespace Domain.Models
{
    public class Room : BaseObject
    {
        public int NumberOfBeds { get; set; }
        public int NumberOfReservedBeds { get; set; }
        public int RoomNumber { get; set; }
        public int PricePerBed { get; set; }
        public string ImageBase64 { get; set; } = string.Empty;
        public IEnumerable<RoomReservation> RoomReservations { get; set; } = new List<RoomReservation>();

    }
}

File: Frontend/src/app/Domain/room.ts
export class Room {
  id: number;
  numberOfBeds: number;
  roomNumber: number;
  pricePerBed: number;
  imageBase64: string;
  numberOfReservedBeds: number;

  constructor(
    id: number,
    numberOfBeds: number,
    roomNumber: number,
    pricePerBed: number,
    imageBase64: string,
    numberOfReservedBeds: number
  ) {
    this.numberOfReservedBeds = numberOfReservedBeds;
    this.id = id;
    this.numberOfBeds = numberOfBeds;
    this.roomNumber = roomNumber;
    this.pricePerBed = pricePerBed;
    this.imageBase64 = imageBase64;
  }
}
